### PR TITLE
fix: TT-141 Have the data persistent in the modal

### PR DESCRIPTION
**Pull request by VanessaIniguez**
Have the data persist in the modal when the modal is closed.

### CODE REVIEW

File: src/app/modules/time-entries/pages/time-entries.component.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export class TimeEntriesComponent implements OnInit, OnDestroy {
   selectedMonth: number;
   selectedYear: number;
   selectedMonthAsText: string;
+  isEdit: boolean;
```
Comment (Angeluz-07) [inline src/app/modules/time-entries/pages/time-entries.component.ts:34]: I think we do not need to use the `isEdit` flag in the time-entries component

File: src/app/modules/shared/components/details-fields/details-fields.component.spec.ts [hunk 2]
```diff
@@ -443,6 +445,20 @@ describe('DetailsFieldsComponent', () => {
...
+    component.isEdit = true;
+    component.ngOnChanges();
+    expect(component.cleanForm).toHaveBeenCalled();
+  });
+
+  it('should persist entry form value', () => {
+    spyOn(component, 'cleanForm');
+    component.isEdit = false;
+    component.ngOnChanges();
+    expect(component.cleanForm).toHaveBeenCalledTimes(0);
```
Comment (Angeluz-07) [inline src/app/modules/shared/components/details-fields/details-fields.component.spec.ts:459]: I think we can use `.not.toHaveBeenCalled()` instead.

**Review by scastillo-jp — APPROVED** [review]

File: src/app/modules/time-entries/pages/time-entries.component.ts [hunk 3]
```diff
@@ -55,41 +52,39 @@ export class TimeEntriesComponent implements OnInit, OnDestroy {
       this.store.dispatch(new entryActions.LoadEntries(this.selectedMonth, this.selectedYear));
     });
   }
-
   newEntry() {
-    this.entry = null;
+    if (this.wasEditingExistingTimeEntry) {
+      this.entry = null;
+    }
+    console.log('Esta es', this.entry);
```
Comment (Angeluz-07) [inline src/app/modules/time-entries/pages/time-entries.component.ts:59]: hehe you missed this console.log.

File: src/app/modules/shared/components/details-fields/details-fields.component.ts [hunk 4]
```diff
@@ -143,7 +143,7 @@ export class DetailsFieldsComponent implements OnChanges, OnInit {
         technology: '',
       });
     } else {
-      this.cleanForm();
+        this.cleanForm();
```
Comment (Angeluz-07) [inline src/app/modules/shared/components/details-fields/details-fields.component.ts:146]: remove whitespace

File: src/app/modules/time-entries/pages/time-entries.component.spec.ts [hunk 5]
```diff
@@ -427,4 +426,18 @@ describe('TimeEntriesComponent', () => {
       expect(entryData.customer_name).toContain('ioet Inc.');
     });
   }));
+
+  it('Should the entry be null if the flag is true', () => {
```
Comment (Angeluz-07) [inline src/app/modules/time-entries/pages/time-entries.component.spec.ts:430]: `should clear the entry If I add a new entry after editing an existing time-entry.`

File: src/app/modules/time-entries/pages/time-entries.component.spec.ts [hunk 6]
```diff
@@ -427,4 +426,18 @@ describe('TimeEntriesComponent', () => {
...
+    component.wasEditingExistingTimeEntry = true;
+    component.newEntry();
+    expect(component.entry).toBe(null);
+  });
+
+  it('Should the input persist the data if the flag is false', () => {
```
Comment (Angeluz-07) [inline src/app/modules/time-entries/pages/time-entries.component.spec.ts:436]: `should persist the input data If I add a new entry and was not editing an existing time-entry.`

File: src/app/modules/time-entries/pages/time-entries.component.ts (hunk 3 above)
Comment (VanessaIniguez) [reply src/app/modules/time-entries/pages/time-entries.component.ts:59]: jajaja Lo siento, ahora eliminaré

File: src/app/modules/time-entries/pages/time-entries.component.spec.ts (hunk 6 above)
Comment (VanessaIniguez) [reply src/app/modules/time-entries/pages/time-entries.component.spec.ts:436]: I consider this test would be doing that if you enter a new entry, the data will persist as long as it is not edited.

**Review by Angeluz-07 — APPROVED** [review]

**Review by israteneda — APPROVED** [review]

**Pull request merged**